python-venv/cmstools/lib/s3/s3.py

```diff
@@ -44,7 +44,7 @@ def s3_client_kwargs() -> JsonDict:
     Decode the S3 credentials from the Kubernetes secret, and return
     the kwargs needed to initialize the boto3 client.
     """
-    secret_data = get_k8s_secret_data(S3_CREDS_SECRET_NAME)
+    secret_data = get_k8s_secret_data(sec_name=S3_CREDS_SECRET_NAME, sec_namespace=K8_CREDS_SECRET_NS)
```

Comment:
I don't think this is necessarily the namespace used for all credentials stored in Kubernetes. This is just the namespace specifically for the S3 credentials. That's why [I suggested using `S3_CREDS_SECRET_NS`](https://github.com/Cray-HPE/cms-tools/pull/318#discussion_r2453120622). And while I know the namespace is a Kubernetes thing, I think it makes more sense to keep that definition in the same place where we define the other S3-related variables.